File: backend-n/src/db/schema.ts
import { sqliteTable, text, integer } from "drizzle-orm/sqlite-core"

export const users = sqliteTable("users", {
  id: integer("id").primaryKey({ autoIncrement: true }),
  username: text("username").notNull().unique(),
  password: text("password").notNull(),
  role: integer("role").notNull().default(0),
  createAt: text("create_at").notNull(),
})

export const refreshTokens = sqliteTable("refresh_tokens", {
  jti: text("jti").primaryKey(),
  userId: integer("user_id")
    .notNull()
    .references(() => users.id),
  userAgent: text("user_agent"),
  fingerprint: text("fingerprint"),
  ipAddress: text("ip_address"),
  lastUsedAt: text("last_used_at").notNull(),
  createAt: text("create_at").notNull(),
  expiresAt: text("expires_at").notNull(),
})

export const settings = sqliteTable("settings", {
  key: text("key").primaryKey(),
  value: text("value").notNull(),
})

File: backend-go/internal/db/schema.go
package db

type User struct {
	ID       int64  `db:"id"`
	Username string `db:"username"`
	Password string `db:"password"`
	Role     int64  `db:"role"`
	CreateAt string `db:"create_at"`
}

type RefreshToken struct {
	Jti         string `db:"jti"`
	UserID      int64  `db:"user_id"`
	UserAgent   string `db:"user_agent"`
	Fingerprint string `db:"fingerprint"`
	IpAddress   string `db:"ip_address"`
	LastUsedAt  string `db:"last_used_at"`
	CreateAt    string `db:"create_at"`
	ExpiresAt   string `db:"expires_at"`
}

type Settings struct {
	Key   string `db:"key"`
	Value string `db:"value"`
}
